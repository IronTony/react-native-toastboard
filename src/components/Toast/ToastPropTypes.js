// @flow
import PropTypes from "prop-types";

export interface ToastProps {
	message: string;
	type: $Keys<typeof ToastType>;
}

export const ToastType = {
	INFO: "INFO",
	ERROR: "ERROR",
	DEBUG: "DEBUG",
	SUCCESS: "SUCCESS",
};

export const ToastPropTypes = {
	message: PropTypes.string.isRequired,
	type: PropTypes.oneOf(Object.keys(ToastType)).isRequired
};
