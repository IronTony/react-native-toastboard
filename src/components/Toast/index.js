// @flow
export * from "./Toast";
export * from "./ToastStyles";
export * from "./ToastPropTypes";
