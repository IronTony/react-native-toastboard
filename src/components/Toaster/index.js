export * from "./Toaster";
export * from "./ToasterStyles";
export * from "./ToasterPropTypes";
