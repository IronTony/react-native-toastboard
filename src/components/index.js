// @flow
export * from "./Toast";
export * from "./Toaster";
