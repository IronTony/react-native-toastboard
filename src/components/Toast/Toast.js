// @flow
import * as React from "react";
import { View, Text } from "react-native";

import { ToastPropTypes } from "./ToastPropTypes";
import { ToastStyles } from "./ToastStyles";

import type { ToastProps } from "./ToastPropTypes";

export const Toast = (props: ToastProps) => (
	<View style={ToastStyles[props.type].container}>
		<Text style={ToastStyles[props.type].message}>
			{props.message}
		</Text>
	</View>
);

Toast.propTypes = ToastPropTypes;
