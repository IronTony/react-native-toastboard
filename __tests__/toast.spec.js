/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";

import { Toast } from "../src/components/Toast";

describe("<Toast />", () => {
    test("Should render correctly", () => {
        const tree = renderer.create(<Toast message="test" type="INFO" />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
