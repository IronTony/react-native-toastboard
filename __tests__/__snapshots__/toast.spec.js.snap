// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> Should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "blue",
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "textAlign": "center",
      }
    }
  >
    test
  </Text>
</View>
`;
